### DISCUSSION TITLE
Fetching ServerInfo on startup while `retry_on_initial_connect` is ON yields 0-values

### DISCUSSION THREAD

**Discussion by barafael** [nats-io/nats.rs#1309] [General]
Asking a server for info obviously only works if the server was previously / is currently connected. Before it was connected, default 0 values are returned. This can lead to problems if you use a maximum payload size of 0 :) 

The solution for this is to wait until an `Event::Connected`, but of course this is not visible in the API. Perhaps it would make sense to expose this fact in the API somehow, for example by returning an Option from server_info? not sure.

**Comment by Jarema** [comment]
Hey. Thanks for filling the issue.

Yes, we could pontentially return `Option` here, though that is of course a breaking change.
Need to think it through. 

Any better ideas @caspervonb ?

**Reply by caspervonb** [reply]
Yeah this is because initially, lazy connection wasn't a thing.
Might want to helpers for waiting for connect etc, e.g `client.connected().await`, this could return the server info.

**Reply by NuSkooler** [reply]
See https://github.com/nats-io/nats.rs/issues/1121
